### PR TITLE
feat(form/builder): add withValidatorHoc to manage the error state of…

### PR DESCRIPTION
**Pull request by AgonisticKatai**
Kapture 2019-06-13 at 11 08 40

### CODE REVIEW

File: components/form/builder/README.md [hunk 1]
```diff
@@ -33,15 +33,49 @@ Defines the form items and dependencies between fields.
...
+    }
   },
   city: {
     type: 'select',
     label: 'Cities',
     placeholder: 'Select a city',
-    errorText: 'Required',
-    next: 'street'
+    next: 'street',
+    empty: {
```
Comment (kikoruiz) [inline components/form/builder/README.md]: ???

File: components/form/builder/src/index.js [hunk 2]
```diff
@@ -8,7 +8,9 @@ import Input from './components/input'
 import TextArea from './components/text-area'
 import AtomButtom from '@schibstedspain/sui-atom-button'
 
-const INITIAL_LIST_ITEM_STATE = []
+import {toCapitalCase} from '@s-ui/js/lib/string'
+
+const INITIAL_LIST_ITEM_STATE = undefined
```
Comment (kikoruiz) [inline components/form/builder/src/index.js]: `const INITIAL_LIST_ITEM_STATE`?

File: components/form/builder/README.md [hunk 3]
```diff
@@ -33,15 +33,49 @@ Defines the form items and dependencies between fields.
...
+    placeholder: 'Type your comments'
+    persists: true,
+    errors: {
+      empty: {
+        text: 'Required'
+      },
+      notAllowed: {
+        character: {
+          text: 'Do not enter special characters',
+          condition: /[`~@#$^&¬|=<>{}[\]\\]/
```
Comment (rmoralp) [inline components/form/builder/README.md]: I prefer "pattern" over "condicion" following W3C https://www.w3.org/TR/2011/WD-html5-20110525/common-input-element-attributes.html#the-pattern-attribute

Comment (rmoralp) [reply components/form/builder/README.md]: Not mandatory, just to think about

File: components/form/builder/src/components/button-group.js [hunk 4]
```diff
@@ -5,27 +5,32 @@ import MoleculeButtonGroup from '@s-ui/react-molecule-button-group'
...
-            </AtomButtom>
-          )
-        })}
-      </MoleculeButtonGroup>
-    </MoleculeField>
-  )
-}
+import WithValidator from '../validatorHoC/WithValidator'
+
+const ButtonGroup = WithValidator(
```
Comment (rmoralp) [inline components/form/builder/src/components/button-group.js]: Just a suggestion. You can improve the readability separating component from HoC, I mean, for example...
`const ButtonGroup = () => {}`
`export default WithValidator(ButtonGroup)`
What do you think?

Comment (AgonisticKatai) [reply components/form/builder/src/components/button-group.js]: DONE!

File: components/form/builder/README.md (hunk 3 above)
Comment (AgonisticKatai) [reply components/form/builder/README.md]: DONE!

File: components/form/builder/README.md (hunk 1 above)
Comment (AgonisticKatai) [reply components/form/builder/README.md]: uppppsss!! sorry! 🤭

Comment (AgonisticKatai) [reply components/form/builder/README.md]: DONE!

File: components/form/builder/src/index.js (hunk 2 above)
Comment (AgonisticKatai) [reply components/form/builder/src/index.js]: DONE!

**Pull request merged**